tanjun/conversion.py

```diff
@@ -253,19 +268,13 @@ def __init__(
             self._allowed_types_repr += f" and {_internal.repr_channel(allowed_types_[-1])}"
 
     @property
-    def async_caches(self) -> collections.Sequence[typing.Any]:
-        # <<inherited docstring from BaseConverter>>.
-        return (_GuildChannelCacheT, _DmCacheT, _ThreadCacheT)
-
-    @property
-    def cache_components(self) -> hikari.api.CacheComponents:
-        # <<inherited docstring from BaseConverter>>.
-        return hikari.api.CacheComponents.GUILD_CHANNELS
-
-    @property
-    def intents(self) -> hikari.Intents:
+    def caches(self) -> collections.Sequence[tuple[typing.Any, hikari.api.CacheComponents, hikari.Intents]]:
         # <<inherited docstring from BaseConverter>>.
-        return hikari.Intents.GUILDS
+        return [
+            (_GuildChannelCacheT, hikari.api.CacheComponents.GUILD_CHANNELS, hikari.Intents.GUILDS),
+            (_DmCacheT, hikari.api.CacheComponents.NONE, hikari.Intents.NONE),
```

Comment:
maybe handle when hikari.api.CacheComponents.NONE means it will always fall back to rest